src/main/java/com/google/cloud/tools/gradle/appengine/core/DownloadCloudSdkTask.java

```diff
@@ -37,35 +42,84 @@ public void setCloudSdkBuilderFactory(CloudSdkBuilderFactory cloudSdkBuilderFact
     this.cloudSdkBuilderFactory = cloudSdkBuilderFactory;
   }
 
-  public void setSdkDownloader(CloudSdkDownloader downloader) {
-    this.downloader = downloader;
+  public void setManagedCloudSdkFactory(ManagedCloudSdkFactory managedCloudSdkFactory) {
+    this.managedCloudSdkFactory = managedCloudSdkFactory;
   }
 
   /** Task entrypoint : Download/update/verify Cloud SDK installation. */
   @TaskAction
-  public void downloadCloudSdkAction() {
+  public void downloadCloudSdkAction() throws Exception {
     String sdkVersion = toolsExtension.getCloudSdkVersion();
-    File sdkHome = toolsExtension.getCloudSdkHome();
-
-    if (sdkHome == null) {
-      if (Strings.isNullOrEmpty(sdkVersion)) {
-        // Wants to download, but version isn't specified; assume latest version
-        sdkVersion = "LATEST";
-      }
-      sdkHome = downloader.downloadSdk(sdkVersion);
-    } else {
-      if (!Strings.isNullOrEmpty(sdkVersion)) {
-        // Sdk home and version specified; validate installation
-        if (!downloader.isSdkValid(sdkVersion, sdkHome)) {
-          throw new GradleException(
-              "Specified Cloud SDK version and actual version of the SDK installed in the "
-                  + "specified directory do not match. You must either specify the correct "
-                  + "cloudSdkHome and cloudSdkVersion, or you can remove the cloudSdkHome field "
-                  + "to download the version you want.");
-        }
-      }
+    managedCloudSdk = managedCloudSdkFactory.newManagedSdk(sdkVersion);
+
+    ProgressListener progressListener =
```

Comment:
I think we probably want to pass these in to this task. (`ProgressListener` and `ConsoleListener`). File a bug to do this in a follow up PR.